client/plots/matrix.interactivity.js

```diff
@@ -554,7 +554,14 @@ function setTermActions(self) {
 				const currentGeneNames = []
 				for (const g of self.config.termgroups) {
 					for (const t of g.lst) {
-						if (t.term.type == 'geneVariant') {
+						// TODO may not need to limit the term type
+						const tmp =
+							self.chartType == 'hierCluster'
+								? // sample counts should be based on clustered samples
+								  ['geneExpression']
```

Comment:
same issue

Reply:
@congyu-lu maybe the fix is to detect the `term.type`s that are in the term `group.type == 'hierCluster'`, and only fill-in the `currentGeneNames` array if the term.type is either `geneExpression` or `geneVariant`

@xzhou82 this means that for clustered term.types that are neither geneExpression or geneVariant, we may not have server-side code that would limit the samples to those that have matching term data (like metabolites). I think `currentGeneNames` is used in that way, to filter the cases from the GDC API, so something equivalent would be needed eventually for other term.types that are clustered?

Reply:
thanks. the `currentGeneNames` was a poor fix. i suggest to simply iterate over g.lst[] and collect any available `gene` or `chr:start-stop` things and put into currentGeneNames array. later will refactor against the array itself